File: src/main/java/no/nav/tjenestepensjon/simulering/service/StillingsprosentResponse.java
package no.nav.tjenestepensjon.simulering.service;

import java.util.List;
import java.util.Map;
import java.util.concurrent.ExecutionException;

import no.nav.tjenestepensjon.simulering.domain.Stillingsprosent;
import no.nav.tjenestepensjon.simulering.domain.TPOrdning;

public class StillingsprosentResponse {
    private final Map<TPOrdning, List<Stillingsprosent>> tpOrdningStillingsprosentMap;
    private final List<ExecutionException> exceptions;

    public StillingsprosentResponse(
            Map<TPOrdning, List<Stillingsprosent>> tpOrdningStillingsprosentMap, List<ExecutionException> exceptions) {
        this.tpOrdningStillingsprosentMap = tpOrdningStillingsprosentMap;
        this.exceptions = exceptions;
    }

    public Map<TPOrdning, List<Stillingsprosent>> getTpOrdningStillingsprosentMap() {
        return tpOrdningStillingsprosentMap;
    }

    public List<ExecutionException> getExceptions() {
        return exceptions;
    }
}

File: src/main/kotlin/no/nav/tjenestepensjon/simulering/v1/service/StillingsprosentResponse.kt
package no.nav.tjenestepensjon.simulering.v1.service

import no.nav.tjenestepensjon.simulering.v1.TPOrdningStillingsprosentMap
import java.util.concurrent.ExecutionException

class StillingsprosentResponse(
        val tpOrdningStillingsprosentMap: TPOrdningStillingsprosentMap,
        val exceptions: List<ExecutionException>
)